main.tf

```diff
@@ -307,15 +314,12 @@ resource "aws_s3_bucket" "origin" {
 }
 
 resource "aws_s3_bucket_public_access_block" "origin" {
-  count                   = (local.create_s3_origin_bucket || local.override_origin_bucket_policy) && var.block_origin_public_access_enabled ? 1 : 0
+  count                   = local.create_s3_origin_bucket || local.override_origin_bucket_policy ? 1 : 0
```

Comment:
Always create this resource, unless we didn't create the origin bucket and we are not overriding its policies.

*NOTE*: I did not test that case